packages/grid/x-data-grid/src/models/api/gridRowApi.ts

```diff
@@ -94,4 +94,15 @@ export interface GridRowApi {
    * @returns {GridRowId[]} The id of each row in the grouping criteria.
    */
   getRowGroupChildren: (params: GridRowGroupChildrenGetterParams) => GridRowId[];
+  /**
+   * Replace a set of rows with new rows.
```

Comment:
I'm not sure. The names are quite self explanatory but still, if we see that people are getting confused we can think of where to add that explanation.

Reply:
But then I'm also wondering. Why is `updateRows` not enough for our use case?

Reply:
The main difference is that `updateRows` does not update the `id` of the row, only the data. I entertained the idea of updating the `updateRows` to handle that use case as well but ultimately decided to have a separate API for it.
PS: the skeleton rows have autogenerated IDs so when a user loads rows that need to replate the skeleton rows their IDs will also be different.

Reply:
Ah right OK. So one updates a row id, the other a row index.

Reply:
I'm not sure about those as well. I agree it's probably best to move the 2 api's under 1 but that we can do once we start v6. For now it would be best to keep the existing one (`updateRows`) and add the new one as `unstable_replaceRows`.

Reply:
> Ah right OK. So one updates a row ID, the other a row index. So maybe we could transpire this in their name grin:

Not exactly
One update the row based on the row id
The other update the row based on the row index